app/src/main/java/com/sdp/movemeet/view/map/GPSRecordingActivity.java

```diff
@@ -104,7 +134,30 @@ public void toggleRecording(View view) {
         recording = !recording;
         recButton.setText(BTN_TEXT_RES.get(recording));
         if (recording) {
+            time = System.currentTimeMillis();
             path.clear();
+        } else {
+            time = System.currentTimeMillis() - time;
+        }
+    }
+
+    @Override
+    public void onBackPressed() {
+        if (!recording) {
+            Intent intent = new Intent(GPSRecordingActivity.this, ActivityDescriptionActivity.class);
```

Comment:
This might interests @jlstoupy98 ! An implementation of the `onBackPressed` method redirecting the user to another activity (here ActivityDescriptionActivity) with some extra information in the intent!